neuralmagic/benchmarks/scripts/benchmark_serving.py

```diff
@@ -273,31 +274,64 @@ def main(args: argparse.Namespace):
     if save_result:
 
         current_dt = datetime.now().strftime("%Y%m%d-%H%M%S")
-        result_json = instantiate_benchmark_results_dict(
-            benchmarking_script_name=Path(__file__).name,
+        result = BenchmarkResult(
+            script_name=Path(__file__).name,
             tensor_parallel_size=args.server_tensor_parallel_size,
             model=args.model,
             tokenizer=args.tokenizer,
             dataset=args.dataset)
-        result_json["date"] = current_dt
-        result_json["script_args"] = vars(args)
-
-        # Populate derived-args for convenience
-        result_json["num_prompts"] = num_prompts
-        result_json["request_rate"] =  \
+        # Populate metadata
+        result[BenchmarkResult.METADATA_KEY_]["date"] = current_dt
+        result[BenchmarkResult.METADATA_KEY_]["script_args"] = vars(args)
+        result[BenchmarkResult.METADATA_KEY_]["num_prompts"] = num_prompts
+        result[BenchmarkResult.METADATA_KEY_]["request_rate"] = \
             request_rate if request_rate < float("inf") else "inf"
-
-        # Merge with benchmark result
-        result_json = {**result_json, **benchmark_result}
+        result[BenchmarkResult.
+               METADATA_KEY_]["duration"] = benchmark_result['duration']
+        result[BenchmarkResult.METADATA_KEY_][
+            "requests_completed"] = benchmark_result['completed']
```

Comment:
it might be better in the long run to hoist the strings out into variables.

Reply:
done.